pkg/azureclient/client.go

```diff
@@ -25,10 +25,10 @@ type Client interface {
 	CreateOrUpdateZone(ctx context.Context, resourceGroupName string, zone string) (dns.Zone, error)
 	DeleteZone(ctx context.Context, resourceGroupName string, zone string) error
 	GetZone(ctx context.Context, resourceGroupName string, zone string) (dns.Zone, error)
-	ListZones(ctx context.Context, resourceGroupName string) (*[]dns.Zone, error)
+	ListZones(ctx context.Context, resourceGroupName string, top *int32) (dns.ZoneListResultPage, error)
 
 	// RecordSets
-	ListRecordSetsByZone(ctx context.Context, resourceGroupName string, zone string) (*[]dns.RecordSet, error)
+	ListRecordSetsByZone(ctx context.Context, resourceGroupName string, zone string, top *int32) (dns.RecordSetListResultPage, error)
```

Comment:
Do not use a pointer for the `top` parameter.

Reply:
Fixed